packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -123,6 +138,36 @@ export const useTreeItem2 = <
       interactions.handleCheckboxSelection(event);
     };
 
+  const createInputHandleKeydown = (otherHandlers: EventHandlers) => (event: any) => {
+    otherHandlers.onKeyDown?.(event);
+    event.stopPropagation();
```

Comment:
Is it needed for all keys or only some specific ones?
Maybe a comment saying which behavior it intends to avoid could help future developer to know when they can remove it

Reply:
When you press a key on the input, the event propagates to the root and activates the keydown handler there and it behaves weirdly -> starts collapsing the item, selecting, etc 😆 
But I removed this and changed the `createRootHandleKeyDown`
![image](https://github.com/user-attachments/assets/521e10a9-6ec1-4db3-9bbc-39d5b9cb1cb7)